File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F12.P2.java
import java.util.OptionalInt;
import java.util.function.IntUnaryOperator;

class Main {
    static int transform() {
        OptionalInt o = OptionalInt.empty();
        // BEGIN PROBE F12.P2
        IntUnaryOperator h = v -> v + 1;
        OptionalInt p = o.stream().map(h).findFirst();
        return p.orElse(0);
        // END PROBE F12.P2
    }

    public static void main(String[] args) {
        System.out.println(transform());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F12.P2.rs
fn mapped(o: Option<i32>) -> i32 {
// BEGIN PROBE F12.P2
let h = |x: i32| x + 1;
let p: Option<i32> = o.map(h);
p.unwrap_or(0)
// END PROBE F12.P2
}

fn main() {
    println!("{}", mapped(None));
}
